src/daft-dsl/src/lit.rs

```diff
@@ -519,29 +521,29 @@ pub fn literals_to_series(values: &[LiteralValue]) -> DaftResult<Series> {
         }
         dtype @ DataType::Timestamp(_, _) => {
             let data = values.iter().map(|lit| unwrap_unchecked!(lit, Timestamp));
-            let physical = Int64Array::from_iter("literal", data);
+            let physical = Int64Array::from_iter(Field::new("literal", DataType::Int64), data);
             TimestampArray::new(Field::new("literal", dtype), physical).into_series()
         }
         dtype @ DataType::Date => {
             let data = values.iter().map(|lit| unwrap_unchecked!(lit, Date));
-            let physical = Int32Array::from_iter("literal", data);
+            let physical = Int32Array::from_iter(Field::new("literal", DataType::Int32), data);
             DateArray::new(Field::new("literal", dtype), physical).into_series()
         }
         dtype @ DataType::Time(_) => {
             let data = values.iter().map(|lit| unwrap_unchecked!(lit, Time));
-            let physical = Int64Array::from_iter("literal", data);
+            let physical = Int64Array::from_iter(Field::new("literal", DataType::Int64), data);
+
             TimeArray::new(Field::new("literal", dtype), physical).into_series()
         }
         DataType::Float64 => {
             let data = values.iter().map(|lit| unwrap_unchecked!(lit, Float64));
 
-            Float64Array::from_iter("literal", data).into_series()
+            Float64Array::from_iter(Arc::new(Field::new("literal", dtype)), data).into_series()
         }
         dtype @ DataType::Decimal128 { .. } => {
             let data = values.iter().map(|lit| unwrap_unchecked!(lit, Decimal));
 
-            let physical = Int128Array::from_iter("literal", data);
-            Decimal128Array::new(Field::new("literal", dtype), physical).into_series()
+            Decimal128Array::from_iter(Arc::new(Field::new("literal", dtype)), data).into_series()
```

Comment:
nit: All of the other `from_iter` calls  in this block are `from_iter::<Field>` but these 2 are `from_iter::<Arc<Field>>`

Reply:
fixed